src/analysis/special_functions/pow.lean

```diff
@@ -907,6 +907,22 @@ begin
     rw [sqrt_eq_zero_of_nonpos h.le, rpow_def_of_neg h, this, cos_pi_div_two, mul_zero] }
 end
 
+lemma rpow_two_sqrt {x : ℝ} (hx : 0 ≤ x): (real.sqrt x)^(2 : ℝ) = x :=
+begin
+  rw [sqrt_eq_rpow, ←rpow_mul hx],
+  norm_num,
+end
+
+@[simp] lemma pow_two_sqrt {x : ℝ} (hx : 0 ≤ x): (real.sqrt x)^2 = x :=
+by rw [←rpow_two, rpow_two_sqrt hx]
```

Comment:
This lemma already exists:

https://github.com/leanprover-community/mathlib/blob/f7e477fdd84a5c43675436be1f6c8d747875e4e6/src/data/real/sqrt.lean#L169-L170

Reply:
Thanks, `library_search` and `simp` weirdly did not find that and also shouldn't there have been a linter error?